### PR TITLE
network: introduce link_get_by_name() and use it in resolving interface name specifier in MultiPathRoute=

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
Rebased. PTAL.

**Comment by keszybz** [comment]
LGTM.

File: src/network/networkd-link.h [hunk 1]
```diff
@@ -210,6 +210,7 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(Link*, link_unref);
 DEFINE_TRIVIAL_DESTRUCTOR(link_netlink_destroy_callback, Link, link_unref);
 
 int link_get(Manager *m, int ifindex, Link **ret);
```
Comment (poettering) [inline src/network/networkd-link.h:212]: maybe remain this function to `link_get_by_ifindex()` and its backing hashmap to `links_by_ifindex`?

**Comment by poettering** [comment]
lgtm, too

**Pull request merged**